README.md

```diff
@@ -2665,24 +2664,39 @@ Other Style Guides
   <a name="semicolons--required"></a><a name="20.1"></a>
   - [21.1](#semicolons--required) **Yup.** eslint: [`semi`](http://eslint.org/docs/rules/semi.html) jscs: [`requireSemicolons`](http://jscs.info/rule/requireSemicolons)
 
+    > Why? When JavaScript encounters a line break without a semicolon, it uses a set of rules called [Automatic Semicolon Insertion](https://tc39.github.io/ecma262/#sec-automatic-semicolon-insertion) to determine whether or not it should regard that line break as the end of a statement, and (as the name implies) place a semicolon into your code before the line break if it thinks so. ASI contains a few eccentric behaviors, though, and your code will break if JavaScript misinterprets your line break. These rules will become more complicated as new features become a part of JavaScript. Explicitly terminating your statements and configuring your linter to catch missing semicolons will help prevent you from encountering issues.
+
     ```javascript
-    // bad
-    (function () {
-      const name = 'Skywalker'
-      return name
-    })()
+    // bad - raises exception
+    const luke = {}
+    const leia = {}
+    [luke, leia].forEach(jedi => jedi.father = 'vader')
+
+    // bad - raises exception
+    const foo = 2
+    ("No! That's impossible!".repeat(foo))
+
+    // bad - returns `undefined` instead of the value on the next line - always happens when `return` is on a line by itself because of ASI!
```

Comment:
adding a semicolon here wouldn't have saved you:

```js
function foo() {
  return
    'search your feelings, you know it to be foo';
}
```

Also, single quotes everywhere instead of double plz :)

Reply:
^ what I mean by this is that this isn't ASI's fault or because of a lack of adding a semicolon. It's not a reason to use or not use a semicolon.

Reply:
As far as whether or not it's a reason to use _semicolons,_ you are correct: the act of a developer adding this semicolon is not, itself, the thing that "would have saved you". However, a linter or IDE that noticed a `return` by itself would note that there was a semicolon missing on the line _containing_ the return. Semicolons would not _directly_ save you, but something warning you that these two lines appeared amiss would.